### PR TITLE
JAMES-2906 Add a message flag updates scenario for a simulation on ma…

### PR DESCRIPTION
**Pull request by Arsnael**
…il corpus

### CODE REVIEW

File: src/main/scala-2.12/org/apache/james/gatling/jmap/scenari/JmapMessageFlagUpdatesScenario.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    val messagesDetailList = "messagesDetailList"
+  }
+
+  def generate(userFeeder: UserFeederBuilder): ScenarioBuilder = {
+    scenario("JmapMessageFlagUpdatesScenario")
+      .feed(userFeeder)
+      .exec(CommonSteps.authentication())
+      .group(MessageFlagUpdates.name)(
+        randomSwitch(
+          70.0 -> exec(JmapMessages.markAsRead()),
```
Comment (chibenwa) [inline src/main/scala-2.12/org/apache/james/gatling/jmap/scenari/JmapMessageFlagUpdatesScenario.scala]: To be calling such an action, one need to have a `messageIds` variable persisted in the session.

This is done by doing a getMessageList before.

You need to add these steps before your flag update random switch:

```
.exec(RetryAuthentication.execWithRetryAuthentication(JmapMailbox.getMailboxes, JmapMailbox.getMailboxesChecks ++ JmapMailbox.saveInboxAs(Keys.inbox)))       

.exec(RetryAuthentication.execWithRetryAuthentication(JmapMessages.listMessages(openpaasListMessageParameters(Keys.inbox)), JmapMessages.listMessagesChecks)))
```

**Review by chibenwa — COMMENTED** [review]
No IT ?

**Review by chibenwa — APPROVED** [review]

**Review by remk — APPROVED** [review]

**Comment by remk** [comment]
```
[error] /home/travis/.sbt/1.0/staging/05cb89211ea66aea052a/gatling-imap/src/main/scala-2.12/com/linagora/gatling/imap/protocol/command/AppendHandler.scala:31:21: value foreach is not a member of java.util.stream.Stream[String]

[error]       content.lines.foreach(executeTextCommand(session))

[error]                     ^

[error] /home/travis/.sbt/1.0/staging/05cb89211ea66aea052a/gatling-imap/src/main/scala-2.12/com/linagora/gatling/imap/protocol/command/AppendHandler.scala:31:47: missing argument list for method executeTextCommand in class AppendHandler

[error] Unapplied methods are only converted to functions when a function type is expected.

[error] You can make this conversion explicit by writing `executeTextCommand _` or `executeTextCommand(_)(_)` instead of `executeTextCommand`.

[error]       content.lines.foreach(executeTextCommand(session))

[error]                                               ^

[error] /home/travis/.sbt/1.0/staging/05cb89211ea66aea052a/gatling-imap/src/main/scala-2.12/com/linagora/gatling/imap/protocol/command/AppendHandler.scala:41:51: value length is not a member of java.util.stream.Stream[String]

[error]       val length = content.length + content.lines.length - 1

[error]                                                   ^

[error] three errors found

[error] (ProjectRef(uri("git://github.com/linagora/gatling-imap.git"), "gatling-imap") / Compile / compileIncremental) Compilation failed
```

**Comment by Arsnael** [comment]
> ```
> [error] /home/travis/.sbt/1.0/staging/05cb89211ea66aea052a/gatling-imap/src/main/scala-2.12/com/linagora/gatling/imap/protocol/command/AppendHandler.scala:31:21: value foreach is not a member of java.util.stream.Stream[String]
> 
> [error]       content.lines.foreach(executeTextCommand(session))
> 
> [error]                     ^
> 
> [error] /home/travis/.sbt/1.0/staging/05cb89211ea66aea052a/gatling-imap/src/main/scala-2.12/com/linagora/gatling/imap/protocol/command/AppendHandler.scala:31:47: missing argument list for method executeTextCommand in class AppendHandler
> 
> [error] Unapplied methods are only converted to functions when a function type is expected.
> 
> [error] You can make this conversion explicit by writing `executeTextCommand _` or `executeTextCommand(_)(_)` instead of `executeTextCommand`.
> 
> [error]       content.lines.foreach(executeTextCommand(session))
> 
> [error]                                               ^
> 
> [error] /home/travis/.sbt/1.0/staging/05cb89211ea66aea052a/gatling-imap/src/main/scala-2.12/com/linagora/gatling/imap/protocol/command/AppendHandler.scala:41:51: value length is not a member of java.util.stream.Stream[String]
> 
> [error]       val length = content.length + content.lines.length - 1
> 
> [error]                                                   ^
> 
> [error] three errors found
> 
> [error] (ProjectRef(uri("git://github.com/linagora/gatling-imap.git"), "gatling-imap") / Compile / compileIncremental) Compilation failed
> ```

@remk : Not sure that looks like an issue from my PR? Seems like @chibenwa got the same on his PR... Probably something wrong with Travis perhaps?

**Comment by chibenwa** [comment]
See linagora/gatling-imap#40

**Pull request merged**